src/Core/Vipr/FileWriter.cs

```diff
@@ -26,8 +31,37 @@ public static void Write(IEnumerable<TextFile> textFilesToWrite, string outputDi
                 if (!Directory.Exists(Path.GetDirectoryName(filePath)))

```

Comment:
Possible Race Condition: Directory.CreateDirectory is thread safe in itself, but if another thread comes in here and tries to check the directory existed, the very same time another thread is creating the directory, and tries to create it again, it will throw an exception. There is no exception handling here to handle that case.